fastserde/avro-fastserde-tests-common/src/test/java/com/linkedin/avro/fastserde/logical/types/LogicalTypesTestBase.java

```diff
@@ -0,0 +1,220 @@
+            classesDir = getCodeGenDirectory();
+            classLoader = URLClassLoader.newInstance(new URL[]{classesDir.toURI().toURL()}, getClass().getClassLoader());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    protected <T extends GenericData> T copyConversions(SpecificData fromSpecificData, T toModelData) {
+        Optional.ofNullable(fromSpecificData.getConversions())
+                .orElse(Collections.emptyList())
+                .forEach(toModelData::addLogicalTypeConversion);
+
+        fixConversionsIfAvro19(toModelData);
+
+        return toModelData;
+    }
+
+    protected <T extends GenericData> void fixConversionsIfAvro19(T modelData) {
+        if (AvroCompatibilityHelperCommon.getRuntimeAvroVersion() == AvroVersion.AVRO_1_9) {
+            // 1. These conversions should be injected by default (avro-compiler issue).
+            // 2. Missing DecimalConversion causes conflict with stringable feature (BigDecimal is considered as string, not bytes)
+            //    details: GenericData.resolveUnion() + SpecificData.getSchemaName().
+            // 3. Time-conversions are needed for GenericDatumWriter/Reader which sees logicalType in the schema
+            //    but is unable to find corresponding converter.
+            // 4. LocalTimestampMillisConversion not available in 1.9.
+
+            List<Conversion<?>> missingConversions = Lists.newArrayList(
+                    new Conversions.DecimalConversion(),
+                    new TimeConversions.TimeMillisConversion(),
+                    new TimeConversions.TimeMicrosConversion(),
+                    new TimeConversions.TimestampMicrosConversion());
+
+            for (Conversion<?> conversion : missingConversions) {
+                modelData.addLogicalTypeConversion(conversion);
+            }
+        }
+    }
+
+    protected <T> byte[] serialize(FastSerializer<T> fastSerializer, T data) throws IOException {
+        InMemoryEncoder encoder = new InMemoryEncoder();
+        fastSerializer.serialize(data, encoder);
+        return encoder.toByteArray();
+    }
+
+    protected <T> byte[] serialize(DatumWriter<T> datumWriter, T data) throws IOException {
+        InMemoryEncoder encoder = new InMemoryEncoder();
+        datumWriter.write(data, encoder);
+
+        return encoder.toByteArray();
+    }
+
+    protected Instant toInstant(Object maybeDate) {
+        if (maybeDate == null) {
+            return null;
+        } else if (maybeDate instanceof LocalDate) {
+            return ((LocalDate) maybeDate).atStartOfDay(ZoneId.systemDefault()).toInstant();
+        } else if (maybeDate instanceof LocalDateTime) {
+            return ((LocalDateTime) maybeDate).toInstant(ZoneOffset.UTC);
+        } else {
+            throw new UnsupportedOperationException(maybeDate + " is not supported (yet)");
+        }
+    }
+
+    /**
+     * Runs various serializers and ensures each one returned the same result.
```

Comment:
The project has a different approach to testing various serializer/deserializer implementations - they pass it as a parameter to the particular test methods, AFAIR. Here you are testing all possible implementations at once. If it's easy to conform to the project's approach then this would result in a better code cohesion.

Reply:
Hm, it looks like the approach has changed.
E.g. `FastSpecificSerializerGeneratorTest.dataAsDecoder(T, org.apache.avro.Schema)` creates serializer in place.

This PR makes fast-serde to support logical types as standard Avro serializers/deserializers do, so comparing their results makes sense. We do not expect any particular result in assertions - all we want is the equality of fast- and standard-Avro (de)serializers.